### PR TITLE
STM32H7: Support the RNG Peripheral

### PR DESCRIPTION
**Pull request by mattofak**
The random number generator on the STM32H7 is identical to the v1 RNG
already present in the library. So use that, and make sure that the RCC
knows about the peripheral.

### CODE REVIEW

File: lib/stm32/h7/Makefile [hunk 1]
```diff
@@ -42,7 +42,7 @@ OBJS += flash_common_all.o flash_common_f.o flash_common_f24.o
 OBJS += fmc_common_f47.o
 OBJS += gpio_common_all.o gpio_common_f0234.o
 OBJS += pwr.o rcc.o
-OBJS += rcc_common_all.o
+OBJS += rcc_common_all.o rng_common_v1.o
```
Comment (karlp) [inline lib/stm32/h7/Makefile:45]: separate lines please, exactly for this reason, it reduces merge conflicts.  only things on same lines are same peripheral

File: lib/stm32/h7/rcc.c [hunk 2]
```diff
@@ -415,3 +420,8 @@ void rcc_set_spi45_clksel(uint8_t clksel) {
 	RCC_D2CCIP1R &= ~(RCC_D2CCIP1R_SPI45SEL_MASK << RCC_D2CCIP1R_SPI45SEL_SHIFT);
 	RCC_D2CCIP1R |= clksel << RCC_D2CCIP1R_SPI45SEL_SHIFT;
 }
+
+void rcc_set_rng_clksel(uint8_t clksel) {
+  RCC_D2CCIP2R &= ~(RCC_D2CCIP2R_RNGSEL_MASK << RCC_D2CCIP2R_RNGSEL_SHIFT);
+  RCC_D2CCIP2R |= clksel << RCC_D2CCIP2R_RNGSEL_SHIFT;
```
Comment (karlp) [inline lib/stm32/h7/rcc.c:426]: whitespace looks wrong here at least

File: lib/stm32/h7/Makefile (hunk 1 above)
Comment (mattofak) [reply lib/stm32/h7/Makefile:45]: Would you like me to split the other lines while we're at it? (Or is it that things related to the same peripheral can be kept together?)

**Comment by karlp** [comment]
this has landed in master now thanks!

**Pull request closed without merging**